### DISCUSSION TITLE
Issue: PI, a reader object cannot find the file selected by a input(type=file) element

### DISCUSSION THREAD

**Discussion by MarcWeustink** [elgatosf/elgato#80] [Help]
The following snipped works on the browsers I tested (FF, edge, chrome), but fails in the PI. It results in a "Error reading file"
Marc
```
<input class="sdpi-item-value" id="select_img_1" type="file" accept=".jpg, .jpeg, .png, .gif, .svg, .webp" onchange="fileChange(this)">
<script>
function fileChange(e) {
const file = e.files[0];
const reader = new FileReader();
reader.onload = function(e) {
	const contents = e.target.result;
};
reader.onerror = function(e) {
	console.error(`Error reading file: ${file.name}`);
};
reader.readAsDataURL(file);

}
</script>
```

**Comment by GeekyEggo** [comment]
Our legacy HTML/JS plugins run in a containerised environment with limited permissions, specifically no file access. The good news - with our new [Node.js SDK](https://github.com/elgatosf/streamdeck) this limitation is removed, and you’re able to access the file system from within the plugin (the limitation are still enforced in the PI).

**Comment by MarcWeustink** [comment]
Since our plugin is a native exe, I've solved it that way. 
When the PI would have read it, a round trip to the plugin wouldn't be necessary
